dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -47,6 +47,7 @@ def reset
     else
       client_state.reset
       reset_session
+      create_storage_id_cookie
```

Comment:
I could be wrong, but I think this means we're creating new storage_ids in a lot of cases where they aren't actually needed (i.e. I go to frozen/reset while logged out, and you've just created a storage_id that I'm likely to never use). I suspect we'd prefer to do a clear without a create new here?

Reply:
I think that the point we should be getting a new storage_id is when we go to the project level in `s/dance`. Before applab-intro, you would never get storage_ids as a logged out user. Applab-intro made it so that you would be able to, but it still wouldn't happen at that high a volume. I would worry if that we start allocating them every time a user starts any script while logged out, we're going to significantly increase the number that we're allocating (with risks that we can't handle that scale). In most cases, that storage_id will not be necessary.

Restricting the allocation to /s/dance would make more sense, but I think the best fix is to always clear on reset, and then let those specific levels that need a storage_id own making sure they have one (something that should already happen I think).

Reply:
okay, that definitely makes sense. so confirming that the solution would be:
- delete `storage_id` in the reset route
- `storage_id` gets created when we are redirected to `/s/dance`, so no need to create a new one

Reply:
no, we don't need to create a new storage id cookie here for this to work. that cookie will be created the first time the signed-out user tries to use a project backed level. I agree with Brent here, we don't want to create this cookie here if we don't have to, so probably it should just be clearing the cookie rather than creating it.

Reply:
sounds good! making that change now

Reply:
> it doesn't seem harmful to create a new storage_id either way

The downside is that it fills up an already large `user_storage_ids` table in our database.